include/gz/sim/MeshInertiaCalculator.hh

```diff
@@ -0,0 +1,230 @@
+    }
+
+    // Calculate integral terms
+    std::vector<double> integral(10);
+    for (int i = 0; i < numTriangles; ++i) {
+        double x0 = triangles[i].v0.X();
+        double y0 = triangles[i].v0.Y();
+        double z0 = triangles[i].v0.Z();
+
+        double x1 = triangles[i].v1.X();
+        double y1 = triangles[i].v1.Y();
+        double z1 = triangles[i].v1.Z();
+
+        double x2 = triangles[i].v2.X();
+        double y2 = triangles[i].v2.Y();
+        double z2 = triangles[i].v2.Z();
+
+        integral[0] += crosses[i].X() * f1[i].X();
+        integral[1] += crosses[i].X() * f2[i].X();
+        integral[2] += crosses[i].Y() * f2[i].Y();
+        integral[3] += crosses[i].Z() * f2[i].Z();
+        integral[4] += crosses[i].X() * f3[i].X();
+        integral[5] += crosses[i].Y() * f3[i].Y();
+        integral[6] += crosses[i].Z() * f3[i].Z();
+        integral[7] += crosses[i].X() * (y0 * g0[i].X() + y1 * g1[i].X() + y2 * g2[i].X());
+        integral[8] += crosses[i].Y() * (z0 * g0[i].Y() + z1 * g1[i].Y() + z2 * g2[i].Y());
+        integral[9] += crosses[i].Z() * (x0 * g0[i].Z() + x1 * g1[i].Z() + x2 * g2[i].Z());
+    }
+
+    for (int i = 0; i < 10; ++i) {
+        integral[i] *= coefficients[i];
+    }
+
+    // Accumulate the result and add it to MassMatrix object of gz::math
+    double volume = integral[0];
+    double mass = volume * density;
+    gz::math::Vector3d center_of_mass = gz::math::Vector3d();
+    center_of_mass.X() = integral[0] / mass;
+    center_of_mass.Y() = integral[1] / mass;
+    center_of_mass.Z() = integral[2] / mass;
+
+    gz::math::Vector3d ixxyyzz = gz::math::Vector3d();
+    gz::math::Vector3d ixyxzyz = gz::math::Vector3d();
+
+    // Diagonal Elements of the Mass Matrix
+    ixxyyzz.X() = integral[5] + integral[6] - mass * (center_of_mass.Y() * center_of_mass.Y() + center_of_mass.Z() * center_of_mass.Z());
+    ixxyyzz.Y() = -(integral[4] + integral[6] - mass * (center_of_mass.Z() * center_of_mass.Z() + center_of_mass.X() * center_of_mass.X()));
+    ixxyyzz.Z() = -(integral[4] + integral[5] - mass * (center_of_mass.X() * center_of_mass.X() + center_of_mass.Y() * center_of_mass.Y()));
+
+    // Off Diagonal Elements of the Mass Matrix
+    ixyxzyz.X() = - (integral[7] - mass * center_of_mass.X() * center_of_mass.Y());
+    ixyxzyz.Y() = - (integral[8] - mass * center_of_mass.Y() * center_of_mass.Z());
+    ixyxzyz.Z() = - (integral[9] - mass * center_of_mass.X() * center_of_mass.Z());
+
+    // Set the values in the MassMatrix object
+    massMatrix.SetMass(mass);
+    massMatrix.SetDiagonalMoments(ixxyyzz);
+    massMatrix.SetOffDiagonalMoments(ixyxzyz);
+    std::cout << "Mass of the mesh is: " << massMatrix.Mass() << std::endl;
+    std::cout << "Diagonal Elements of the Inertia Matrix are: " << massMatrix.DiagonalMoments() << std::endl;
+    std::cout << "Off Diagonal Elements of the Inertia Matrix are: " << massMatrix.OffDiagonalMoments() << std::endl;
```

Comment:
use `gzdbg` or remove

Reply:
Removed in [c79a397](https://github.com/gazebosim/gz-sim/pull/2061/commits/c79a397140370dff5c9fa65aa73d7ce321b4cd9b)